beaconChain/lib/powChain/index.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers, Contract } from 'ethers';
+import { EtherscanProvider } from 'ethers/providers';
+import { Deposit, DepositData, DepositInput } from '../../interfaces/blocks';
+import { ChainStart } from "../../interfaces/state";
+import { MAINNET_DEPOSIT_ADDRESS, DEPOSIT_CONTRACT_ADDRESS, DEPOSIT_CONTRACT_BLOCK_NUMBER, DEPOSIT_CONTRACT_ABI } from "../../constants/constants";
+
+// Type stubs
+type uint64 = number;
+type hash32 = string;
+
+
+// NOTE: This is stubbed
+// TODO: We should find a way to reject the promise somehow.
+// TODO: There is porbably a better way to scrape the logs.
+/**
+ * Polls the eth1.x deposit contract for validator deposits. If the ChainStart log is emitted, return the promise with
+ * intial state data.
+ * @returns {Promise<ChainStart>}
+ */
+const waitForChainStart = () => {
```

Comment:
I think its plausible to test this functionality at this point, the notion of waiting for the ChainStart log isn't going to change. Please add tests if you agree.

Reply:
You know you're actually right. I'll probably haven't to import the vyper contract? Unless #66 gets closed

Reply:
You know you're actually right. I'll probably haven't to import the vyper contract? Unless #66 gets closed

Reply:
#66 has been closed

Reply:
Do we want to add truffle/embark? Before continuing we'll need to discuss if we want a framework for testing smart contracts or not

Reply:
What about etherlime? It uses ethers.

Reply:
etherlime looks good. Still need to figure out what we want to do. Ultimately, we could do it manually.